File: api/models/check_in.py
from uuid import uuid4

from django.db import models


class CheckIn(models.Model):
    id = models.UUIDField(default=uuid4, primary_key=True, unique=True, editable=False)
    created = models.DateTimeField(auto_now_add=True)
    modified = models.DateTimeField(auto_now=True)
    duration = models.FloatField()
    start_time = models.TimeField()
    record_date = models.DateField()
    tag = models.CharField(max_length=64)
    activities = models.CharField(max_length=256)

    class Meta:
        ordering = ["-record_date", "-start_time"]
        verbose_name = "Check in"
        verbose_name_plural = "Check ins"

    def __str__(self):
        hour = "hr" if self.duration == 1 else "hrs"
        return f"{self.duration} {hour} #{self.tag} {self.activities}"

File: api/models/checkin.go
package models

type CreateCheckinInput struct {
    Duration   float64 `json:"duration" validate:"required,gt=0"`
    StartTime  string  `json:"start_time" validate:"required"`
    RecordDate string  `json:"record_date" validate:"required"`
    Tag        string  `json:"tag" validate:"required"`
    Activities string  `json:"activities" validate:"required"`
}

type ListCheckinsInput struct {
    Page      int32  `json:"page" validate:"gte=0"`
    StartDate string `json:"start_date" validate:"datetime"`
    EndDate   string `json:"end_date" validate:"datetime"`
    Tag       string `json:"tag"`
}

type ListAllCheckinsInput struct {
    StartDate string `json:"start_date" validate:"datetime"`
    EndDate   string `json:"end_date" validate:"datetime"`
}
